include/Utils/DIPUtils.h

```diff
@@ -22,6 +22,7 @@
 #ifndef INCLUDE_UTILS_DIPUTILS_H
 #define INCLUDE_UTILS_DIPUTILS_H
 
+#include "DIP/DIPOps.h"
```

Comment:
Why are you including this file?

Reply:
Sorry, my `clangd` automatically includes this file but I haven't noticed it. It is unnecessary for compilation, but removing it will cause my `clangd` to produce a ton of errors for this file.

The real problem is that the `Utils/Utils.h` file is not self-contained. `Utils/Utils.h` is which needs this inclusion. I will remove this #include and add one to `Utils/Utils.h` in another PR.

Reply:
Can you please share the error log you get when you don't include that file.

Reply:
![Screenshot from 2022-11-11 23-19-13](https://user-images.githubusercontent.com/29816865/201370408-d1e1585f-c30c-4c7f-825b-cf1e3921cbd4.png)

Reply:
I think this is an issue related only to intellisense, as [this](https://github.com/meshtag/buddy-mlir/actions/runs/3442213404/jobs/5742558798) run compiled well.  
(Ref : https://stackoverflow.com/questions/31943634/visual-studio-2015-or-2017-shows-intellisense-errors-but-solution-compiles )

Reply:
> Sorry, my `clangd` automatically includes this file but I haven't noticed it. It is unnecessary for compilation, but removing it will cause my `clangd` to produce a ton of errors for this file.

Yes, you are right. It's **only** an IntelliSense error in **`clangd`**, not about using **`clang`** to compile the whole project. I can also successfully compile the project without that `#include` locally as well.

> The real problem is that the `Utils/Utils.h` file is not self-contained. `Utils/Utils.h` is which needs this inclusion. I will remove this #include and add one to `Utils/Utils.h` in another PR.

But the IntelliSense error indicates a possible problem in `Utils/Utils.h`. A header file that is not self-contained will depend on the order of inclusion in other files. For example:

```cpp
// in file A.cpp that needs to include Utils.h
// this will work
#include "DIP/DIPOps.h"
#include "Utils/Utils.h"

// in file B.cpp that needs to include Utils.h
// this will NOT work
#include "Utils/Utils.h"
#include "DIP/DIPOps.h"
```

As we include the `Utils/Utils.h` at the beginning of `Utils/DIPUtils.h`, we also "pollute" the `Utils/DIPUtils.h`, making it not self-contained as well. In fact, the whole project can compile is only because we have not written any C++ file (non-header file) that will include `Utils/DIPUtils.h` as the first file it will include.

In my opinion, making header files self-contained should be a best practice for C/C++ projects. So I said I may open another PR to fix this (or maybe more non self-contained header files in the project). You may also refer [this SO answer](https://stackoverflow.com/a/1892054).

Anyway, this problem is **not** related to the goal of the current PR (fixing the compilation failure under `clang`), we may discuss it later and I also push a new commit to delete the unnecessary inclusion. I am sorry if my previous words were not clear enough.

Reply:
Alright